internal/pkg/cli/pipeline_update.go

```diff
@@ -197,6 +210,36 @@ func (o *updatePipelineOpts) deployPipeline(in *deploy.CreatePipelineInput) erro
 	return nil
 }
 
+func (o *updatePipelineOpts) ConnectionName() (string, error) {
+	if o.repoURL == "" {
+		return "", fmt.Errorf("unable to locate the repository URL")
+	}
+	match := bbRepoExp.FindStringSubmatch(o.repoURL)
+	if len(match) == 0 {
+		return "", fmt.Errorf("unable to parse the repository URL")
+	}
+	matches := make(map[string]string)
+	for i, name := range bbRepoExp.SubexpNames() {
+		if i != 0 && name != "" {
+			matches[name] = match[i]
+		}
+	}
+	owner := matches["owner"]
+	repo := matches["repo"]
+
+	const (
+		ownerLetters = 5
+		repoLetters  = 18
+	)
+	if len(owner) > ownerLetters {
+		owner = owner[:ownerLetters]
+	}
+	if len(repo) > repoLetters {
+		repo = repo[:repoLetters]
+	}
+	return fmt.Sprintf("copilot-%s-%s", owner, repo), nil
```

Comment:
What do you think defining this format pattern as a const so that it would be easier to locate how the connection is formatted?